managed/CounterStrikeSharp.API/Modules/Commands/Targeting/Target.cs

```diff
@@ -80,48 +83,36 @@ public Target(string target)
-                return player.IsBot;
-            case TargetType.GroupHumans:
-                return !player.IsBot && !player.IsHLTV;
-            case TargetType.GroupAlive:
-                return player.PlayerPawn is { IsValid: true, Value.LifeState: (byte)LifeState_t.LIFE_ALIVE };
-            case TargetType.GroupDead:
-                return player.PlayerPawn is { IsValid: true, Value.LifeState: (byte)LifeState_t.LIFE_DEAD or (byte)LifeState_t.LIFE_DYING };
-            case TargetType.GroupNotMe:
-                return player.SteamID != caller?.SteamID;
-            case TargetType.PlayerMe:
-                return player.SteamID == caller?.SteamID;
-            case TargetType.IdUserid:
-                return player.UserId.ToString() == Slug;
-            case TargetType.IdSteamEscaped:
-                return ((SteamID)player.SteamID).SteamId2 == Slug;
-            case TargetType.IdSteam64:
-                return ((SteamID)player.SteamID).SteamId64.ToString() == Slug;
-            case TargetType.ExplicitName:
-            case TargetType.ImplicitName:
-                return player.PlayerName.Contains(Slug, StringComparison.OrdinalIgnoreCase);
-            default:
-                return false;
-        }
+            TargetType.PlayerAim => caller != null && player == gameRules.GetClientAimTarget(caller),
+            TargetType.TeamCt => player.Team == CsTeam.CounterTerrorist,
+            TargetType.TeamT => player.Team == CsTeam.Terrorist,
+            TargetType.TeamSpec => player.Team == CsTeam.Spectator,
+            TargetType.GroupAll => !player.IsHLTV,
+            TargetType.GroupBots => player.IsBot,
+            TargetType.GroupHumans => !player.IsBot && !player.IsHLTV,
+            TargetType.GroupAlive => player.PlayerPawn is { IsValid: true, Value.LifeState: (byte)LifeState_t.LIFE_ALIVE },
+            TargetType.GroupDead => player.PlayerPawn is { IsValid: true, Value.LifeState: (byte)LifeState_t.LIFE_DEAD or (byte)LifeState_t.LIFE_DYING },
+            TargetType.GroupNotMe => player.SteamID != caller?.SteamID,
+            TargetType.PlayerMe => player.SteamID == caller?.SteamID,
+            TargetType.IdUserid => player.UserId.ToString() == Slug,
+            TargetType.IdSteamEscaped => ((SteamID)player.SteamID).SteamId2 == Slug,
+            TargetType.IdSteam64 => ((SteamID)player.SteamID).SteamId64.ToString() == Slug,
+            TargetType.ExplicitName => player.PlayerName.Contains(Slug, StringComparison.OrdinalIgnoreCase),
+            TargetType.ImplicitName => player.PlayerName.Contains(Slug, StringComparison.OrdinalIgnoreCase),
+            _ => false
+        };
     }
 
     public TargetResult GetTarget(CCSPlayerController? caller)
     {
-        var players = Utilities.GetPlayers().Where(player => TargetPredicate(player, caller)).ToList();
-        
+        CCSGameRules? gameRules = Utilities.FindAllEntitiesByDesignerName<CCSGameRulesProxy>("cs_gamerules").FirstOrDefault()?.GameRules;
```

Comment:
I believe we should not get gamerules each time here, might worth using the `EntityCache<T>` from discord as a built in type? not sure, just a first guess

Reply:
Can you give me an example?

Reply:
but with that, it would only get gamerules if its invalid, otherwise it would use the cached value instead of getting it each time that method is called which is more expensive than it should be

Reply:
also you would still need to check for null/not valid